### ISSUE TITLE
Integrate traditional Mongolian (mongol bichig) in ibus-table

### ISSUE THREAD

**Issue by Popolon** [kaio/ibus-table#72]
I made few years ago a [traditional Mongolian input method for ibus-table](https://framagit.org/popolon/ibus-table-mongol_bichig) also called uyghur script as it is derived from ancient uyghur script from the Chingis Khagan time, they use arabic script in China today. This script mainly used by Mongolians people and Mongolian speakers of China. This currently is only complete with Latin based keyboard as used in China. The support of Cyrillic version is incomplete, I miss good reference for it, but it allow at least to type words, it could be useful to for Mongolian people of Mongolia and Russia that mainly use Cyrillic keyboard.

There are two special characters used two links suffixes that are not displayed, corresponding to F and G capital keys, as there is no capitals in this script.

I believe it should be integrated to have better visibility among them. The growing integration of Linux distribution in phone and tablets in China, will also help them having an input method by default. This is based on [Almas Mongolian Keyboard Layout](http://www.mongolfont.com/en/misc/mnglkbd.html), a multiplatform standard for Mongolian in China.

I agree to change the license as needed to match ibus-table requirement.

**Comment by Popolon** [comment]
Searching for an asciiart emoji table I used long time ago, I just found there was another implementation, by chinese author, it could be more finished implementation : https://github.com/hfl/traditional-mongolian

**Comment by mike-fabian** [comment]
So the table at https://github.com/hfl/traditional-mongolian is not good for you (you write you never managed to use it) and your own table is better, do I understand that right?

**Comment by mike-fabian** [comment]
https://framagit.org/popolon/ibus-table-mongol_bichig/-/blob/master/mongol_bichig.txt#L141 contains:

```
Q	ZWNJ	0
```

I guess this is supposed to insert the character U+200C ZERO WIDTH NON-JOINER but when using your table, it doesn’t do that, it inserts the string “ZWNJ”. This is a mistake, right?

**Comment by mike-fabian** [comment]
Other mistakes:
```
?       NNBSP   0
```

Probably U+202F NARROW NO-BREAK SPACE

```
v       MVS     0
```

Probably U+180E ᠎ MONGOLIAN VOWEL SEPARATOR

```
п       FVS1    0
```

Probably U+180B MONGOLIAN FREE VARIATION SELECTOR ONE

```
Г       FVS2    0
```

Probably U+180C MONGOLIAN FREE VARIATION SELECTOR TWO

```
Ш       FVS3    0
```
Probably U+180D MONGOLIAN FREE VARIATION SELECTOR THREE

```
Ф       SWNJ    0
```

Misspelling of ZWNJ? Should probably be U+200C ZERO WIDTH NON-JOINER?

**Comment by mike-fabian** [comment]
http://www.mongolfont.com/en/misc/mnglkbd.html shows the Mongolian characters on a US and a JP keyboard layout.

Your table contains:

```
### The Keyboard Layout used by this table. Unset or set to "Unknown" to accept any kind of la
youts.
LAYOUT = us
```

which forces a switch to the US keyboard layout when switching to your input method.

I think it would be better to use

```
LAYOUT = default
```

to keep the layout which wasx set before switching to your table.

Because there are probably many keyboard layouts which would work with your table, not only US and JP.
Some very different keyboards like the French AZERTY might not work so well but it looks like your table would work for a lot of keyboard layouts.

On top of that, your table contains the (unfinished you say) cyrillic part.

If you force the US keyboard layout, the cyrillic part cannot be used as you cannot input cyrillic characters with the US layout.

But if you use 

```
LAYOUT = default
```

Then one could switch to a US layout and then to your table and use the section of your table with Latin letters or switch to a RU layout and then switch to your table to use the cyrillic section of your table.

**Comment by Popolon** [comment]
> So the table at https://github.com/hfl/traditional-mongolian is not good for you (you write you never managed to use it) and your own table is better, do I understand that right?

It's probably another method I don't know and so never managed to use, perhaps a method well known in some part of China, I believe I exchanged with him at this time and it was in Xinjiang where Oirat/Kalmyk mongolian is used (Used in Eastern Europe in kalmykia inside russian federation too).


It looks like you are right for the extended characters and the layout. It worked as is on my french AZERTY keyboard and as it is phonetic mapping that's easy enough to be used. I used only for few research on Mongolian names on some dictionaries/documents/Wikipedia for resources about Mongolians history, and for a simple basic Stardict format dictionary I made (French <->  Cyrillic Mongolian (in Khakha mongolian, as in  Mongolia) + traditional Mongolian, as in Inner Mongolia, and history) And a non-latin=> latin dictionnary (with cyrillic, mongol, and kanas). I don't master this script. I just hope Mongolians users could have one too, and get feed back if needed. I don't remember how was used these few special characters, but only one for suffixes. Characters are now widely available in Google fonts, and both GNU/Linux and Android generally manage well the changes to do (some characters shapes change depending on their own position in words).

**Comment by mike-fabian** [comment]
There are some differences between http://www.mongolfont.com/en/misc/mnglkbd.html and your table.
For example, your table has:

```
A	ᠩ	0
```
ᠩ U+1829 MONGOLIAN LETTER ANG

but http://www.mongolfont.com/en/misc/mnglkbd.html has MVS on A.

**Comment by mike-fabian** [comment]
Is it OK if I fix your table to agree **exactly** with http://www.mongolfont.com/en/misc/mnglkbd.html ?

**Comment by mike-fabian** [comment]
Your table has:

```
o	ᠣ	0
```

which is U+1823` MONGOLIAN LETTER O

But http://www.mongolfont.com/en/misc/mnglkbd.html puts ᠥ U+1825 MONGOLIAN LETTER OE on the key o.

**Comment by mike-fabian** [comment]
Your table:

```
w	ᠸ	0
```
U+1838 MONGOLIAN`` LETTER WA
But http://www.mongolfont.com/en/misc/mnglkbd.html  has U+1823` MONGOLIAN LETTER O on w

**Comment by mike-fabian** [comment]
I fixed your table to make it agree completely with http://www.mongolfont.com/en/misc/mnglkbd.html

Here is the improved file:

[mongol_bichig.txt](https://github.com/kaio/ibus-table/files/6552494/mongol_bichig.txt)

Here as a patch:

(I created an account on https://framagit.org/ but the account has not been approved yet, as soon as it is approved I can send you a merge request).

```
From 86df08a1f4e8fe46751fa8af267a5a335308dc70 Mon Sep 17 00:00:00 2001
From: Mike FABIAN <mfabian@redhat.com>
Date: Thu, 27 May 2021 10:39:48 +0200
Subject: [PATCH] Fix to agree exactly with
 http://www.mongolfont.com/en/misc/mnglkbd.html

- Use the real characters instead of ASCII descriptions like MVS, FVS1, ...
- Fix all other mistakes where there were differences to
  http://www.mongolfont.com/en/misc/mnglkbd.html
- Add comments with the Unicode code points and character names
  (especially useful for the invisible characters like MVS, FVS1, ...
  helps a bit when editing this table, the comments say what should be there)
- Change SYMBOL and STATUS_PROMPT, a cyrillic letter seemed wrong there
- Change LAYOUT to default, this table is usable with many layouts
- DYNAMIC_ADJUST=FALSE, this table is a simple transliteration table,
  no table of candidates is shown, adjusting user frequencies of candidates
  is thus useless.
- MAX_KEY_LENGTH=1 (There are no longer keys in this table)
- Fix VALID_INPUT_CHARS (the cyrillic characters should be added there when
  the cyrillic part of the table is completed
- SELECT_KEYS = F1,F2,F3,F4,F5,F6,F7,F8,F9,F10 (Select keys are not needed anyway
  because this table doesn't use a candidate list. But as the digits
  0, 1, ...,9 are used by this table to input Mongolian digits, not
  specifying them in SELECT_KEYS is clearer.
- Add USER_CAN_DEFINE_PHRASE = FALSE and empty RULES (just for documentation)
---
 mongol_bichig.txt | 300 ++++++++++++++++++++++++++--------------------
 1 file changed, 169 insertions(+), 131 deletions(-)

diff --git a/mongol_bichig.txt b/mongol_bichig.txt
index 0401501..49fa329 100644
--- a/mongol_bichig.txt
+++ b/mongol_bichig.txt
@@ -18,10 +18,23 @@ UUID = 1a92f9dd-f4fe-41f3-8a3d-71bd51e911e8
 ### A unique number indicates the version of this file.
 ### For example the last modified date of this file.
 ### This number must be less than 2^32.
-SERIAL_NUMBER = 20150105
+SERIAL_NUMBER = 20210526
 
 ICON = mongol_bichig.svg
 
+### The symbol to be displayed in IM switchers
+### This is used in the Gnome panel!
+### unfortunately Gnome3 has problems displaying “☑<text>” and “☐<text>”
+### correctly in the status bar when <text> is longer than two ASCII characters.
+### (the checkbox becomes invisible making it impossible to see whether
+### the input is active or not).
+### I.e. in case of Mongolian, either use “mn” or if using Mongolian characters
+### use only 1 character to make sure it is not cut off.
+SYMBOL = ᠠ
+
+### Prompt string to be displayed in the status area.
+STATUS_PROMPT = ᠠ
+
 ### The default name of this table
 NAME = Mongol bichig
 
@@ -35,11 +48,13 @@ LANGUAGES = mn_MN, mvf_CN
 ### The author of this table
 AUTHOR = Popolon <popolon@popolon.org>
 
-### Prompt string to be displayed in the status area.
-STATUS_PROMPT = Ө
-
-### The Keyboard Layout used by this table. Unset or set to "Unknown" to accept any kind of layouts.
-LAYOUT = us
+### The Keyboard Layout used by this table.
+### Set to "default" to accept any kind of layout
+###
+### This table can use Latin characters or Cyrillic characters as input,
+### it can be used with most keyboard layouts using
+### Latin or Cyrillic characters.
+LAYOUT = default
 
 ### If true then the first candidate phrase
 ### will be selected automatically during inputing.
@@ -57,7 +72,7 @@ AUTO_COMMIT = TRUE
 AUTO_SPLIT = TRUE
 
 ### If true then the phrases' frequencies will be adjusted dynamically.
-DYNAMIC_ADJUST = TRUE
+DYNAMIC_ADJUST = FALSE
 
 ### If true then the preedit area will be filled up by the current candidate phrase automatically.
 AUTO_FILL = TRUE
@@ -79,10 +94,10 @@ USE_FULL_WIDTH_LETTER = FALSE
 DEF_FULL_WIDTH_LETTER = FALSE
 
 ### The maxmium length of a key.
-MAX_KEY_LENGTH = 2
+MAX_KEY_LENGTH = 1
 
 ### Valid input chars.
-VALID_INPUT_CHARS = ABWGDE&VZIJKLMNOPQRSTUFHCYXabwgdevzijklmnopqrstufhcyx"'[]{}();:,.<>/?-_+$\|`~
+VALID_INPUT_CHARS = 1234567890ACDEFGHIKLNOQRSTWZabcdefghijklmnopqrstuvwxyz{}&'/:
 
 ### Single wildcard char, can have multiple chars.
 ###SINGLE_WILDCARD_CHAR = ?
@@ -100,7 +115,7 @@ COMMIT_KEYS = space
 FORWARD_KEYS = Return
 
 ### The key strokes to select candidiate phrases.
-SELECT_KEYS = 1,2,3,4,5,6,7,8,9
+SELECT_KEYS = F1,F2,F3,F4,F5,F6,F7,F8,F9,F10
 
 ### The key strokes to page up the lookup table.
 PAGE_UP_KEYS = Page_Up
@@ -108,129 +123,152 @@ PAGE_UP_KEYS = Page_Up
 ### The key strokes to page down the lookup table.
 PAGE_DOWN_KEYS = Page_Down
 
+### Whether the user is allowed to define a phrase, default is true
+### You have to define the word construction rules below.
+### For input methods which do not input phrases, set this to False
+USER_CAN_DEFINE_PHRASE = FALSE
+
+### Rules for constructing user defined phrase (This is not useful
+### for this table, therefore it is set to the empty string.)
+RULES =
+
 END_DEFINITION
 
 ### Begin Table data.
+###
+### For the Latin part of the table see the keyboard layouts
+### at http://www.mongolfont.com/en/misc/mnglkbd.html
 BEGIN_TABLE
-1	᠑	0
-2	᠒	0
-3	᠓	0
-4	᠔	0
-5	᠕	0
-6	᠖	0
-7	᠗	0
-8	᠘	0
-9	᠙	0
-0	᠐	0
-A	ᠩ	0
-B		0
-C	ᡂ	0
-D		0
-E	ᠧ	0
-F	᠍	0
-G	᠌	0
-H	ᠾ	0
-I		0
-J		0
-K	ᠻ	0
-L	ᡀ	0
-M		0
-N		0
-O	ᠥ	0
-P		0
-Q	ZWNJ	0
-R	ᠿ	0
-S		0
-T	₮	0
-U	ᠤ	0
-V		0
-W		0
-X		0
-Y		0
-Z	ᡁ	0
-{		0
-}		0
-<	《	0
->	》	0
-:	᠄	0
-"	᠁	0
-?	NNBSP	0
-?	？	0
-a	ᠠ	0
-b	ᠪ	0
-c	ᠼ	0
-d	ᠳ	0
-e	ᠡ	0
-f	ᠹ	0
-g	ᠭ	0
-h	ᠬ	0
-i	ᠢ	0
-j	ᠵ	0
-k	ᠺ	0
-l	ᠯ	0
-m	ᠮ	0
-n	ᠨ	0
-o	ᠣ	0
-p	ᠫ	0
-q	ᠴ	0
-r	ᠷ	0
-s	ᠰ	0
-t	ᠲ	0
-u	ᠦ	0 ?ue, pas u , ni oe
-v	MVS	0
-w	ᠸ	0
-x	ᠱ	0
-y	ᠶ	0
-z	ᠽ	0
-ф	ᠹ	0
-ц	ᠴ	0
-у	ᠤ	0
-ж	ᠵ	0
-э	ᠡ	0
-н	ᠨ	0
-г	ᠭ	0
-ш	ᠱ	0
-ү	ᠦ	0
-з	ᠽ	0
-к	ᠻ	0
-ъ	ᠫ	0
-й	ᠩ	0
-ы	MVS	0
-б	ᠪ	0
-ө	ᠥ	0
-а	ᠠ	0
-х	ᠬ	0
-р	ᠷ	0
-о	ᠣ	0
-л	ᠯ	0
-д	ᠳ	0
-п	FVS1	0
-я	ᠶ	0
-ч	ᠼ	0
-ё	ᠧ	0
-с	ᠰ	0
-м	ᠮ	0
-и	ᠢ	0
-т	ᠲ	0
-ь	᠂	0
-в	᠃	0
-ю	NNBSP	0
-Б	ᠸ	0
-Р	ᠿ	0
-Т	᠅	0
-И	᠀	0
-Ь	《	0
-В	》	0
-Х	ᠾ	0
-Ч	ᡂ	0
-Я	ᡁ	0
-Л	ᡀ	0
-О	ᠺ	0
-Г	FVS2	0
-Ш	FVS3	0
-Ф	SWNJ	0
-У	【	0
-Ж	】	0
-Э	〈	0
-Н	〉	0
+1	᠑	0	### U+1811 MONGOLIAN DIGIT ONE
+2	᠒	0	### U+1812 MONGOLIAN DIGIT TWO
+3	᠓	0	### U+1813 MONGOLIAN DIGIT THREE
+4	᠔	0	### U+1814 MONGOLIAN DIGIT FOUR
+5	᠕	0	### U+1815 MONGOLIAN DIGIT FIVE
+6	᠖	0	### U+1816 MONGOLIAN DIGIT SIX
+7	᠗	0	### U+1817 MONGOLIAN DIGIT SEVEN
+8	᠘	0	### U+1818 MONGOLIAN DIGIT EIGHT
+9	᠙	0	### U+1819 MONGOLIAN DIGIT NINE
+0	᠐	0	### U+1810 MONGOLIAN DIGIT ZERO
+A	᠎	0	### U+180E MONGOLIAN VOWEL SEPARATOR
+### B not used
+C	ᡂ	0	### U+1842 MONGOLIAN LETTER CHI
+D	᠋	0	### U+180B MONGOLIAN FREE VARIATION SELECTOR ONE
+E	ᠧ	0	### U+1827 MONGOLIAN LETTER EE
+F	᠌	0	### U+180C MONGOLIAN FREE VARIATION SELECTOR TWO
+G	᠍	0	### U+180D MONGOLIAN FREE VARIATION SELECTOR THREE
+H	ᠾ	0	### U+183E MONGOLIAN LETTER HAA
+I	᠊	0	### U+180A MONGOLIAN NIRUGU
+### J not used
+K	ᠻ	0	### U+183B MONGOLIAN LETTER KHA
+L	ᡀ	0	### U+1840 MONGOLIAN LETTER LHA
+### M not used
+N	ᠩ	0	### U+1829 MONGOLIAN LETTER ANG
+O	ᠥ	0	### U+1825 MONGOLIAN LETTER OE
+### P not used
+Q	᠁	0	### U+1801 MONGOLIAN ELLIPSIS
+R	ᠿ	0	### U+183F MONGOLIAN LETTER ZRA
+S	 	0	### U+202F NARROW NO-BREAK SPACE
+### T is not used in http://www.mongolfont.com/en/misc/mnglkbd.html)
+### so we can put the currency symbol for the Mongolian currency there:
+### https://en.wikipedia.org/wiki/Mongolian_t%C3%B6gr%C3%B6g
+T	₮	0	### U+20AE TUGRIK SIGN
+### U not used
+### V not used
+W	ᠸ	0	### U+1838 MONGOLIAN LETTER WA
+### X not used
+### Y not used
+Z	ᡁ	0	### U+1841 MONGOLIAN LETTER ZHI
+a	ᠠ	0	### U+1820 MONGOLIAN LETTER A
+b	ᠪ	0	### U+182A MONGOLIAN LETTER BA
+c	ᠼ	0	### U+183C MONGOLIAN LETTER TSA
+d	ᠳ	0	### U+1833 MONGOLIAN LETTER DA
+e	ᠡ	0	### U+1821 MONGOLIAN LETTER E
+f	ᠹ	0	### U+1839 MONGOLIAN LETTER FA
+g	ᠭ	0	### U+182D MONGOLIAN LETTER GA
+h	ᠬ	0	### U+182C MONGOLIAN LETTER QA
+i	ᠢ	0	### U+1822 MONGOLIAN LETTER I
+j	ᠵ	0	### U+1835 MONGOLIAN LETTER JA
+k	ᠺ	0	### U+183A MONGOLIAN LETTER KA
+l	ᠯ	0	### U+182F MONGOLIAN LETTER LA
+m	ᠮ	0	### U+182E MONGOLIAN LETTER MA
+n	ᠨ	0	### U+1828 MONGOLIAN LETTER NA
+o	ᠥ	0	### U+1825 MONGOLIAN LETTER OE
+p	ᠫ	0	### U+182B MONGOLIAN LETTER PA
+q	ᠴ	0	### U+1834 MONGOLIAN LETTER CHA
+r	ᠷ	0	### U+1837 MONGOLIAN LETTER RA
+s	ᠰ	0	### U+1830 MONGOLIAN LETTER SA
+t	ᠲ	0	### U+1832 MONGOLIAN LETTER TA
+u	ᠦ	0	### U+1826 MONGOLIAN LETTER UE ?ue, pas u , ni oe
+v	ᠤ	0	### U+1824 MONGOLIAN LETTER U
+w	ᠣ	0	### U+1823 MONGOLIAN LETTER O
+x	ᠱ	0	### U+1831 MONGOLIAN LETTER SHA
+y	ᠶ	0	### U+1836 MONGOLIAN LETTER YA
+z	ᠽ	0	### U+183D MONGOLIAN LETTER ZA
+### Punctuation:
+{	《	0	### U+300A LEFT DOUBLE ANGLE BRACKET
+}	》	0	### U+300B RIGHT DOUBLE ANGLE BRACKET
+### The table on http://www.mongolfont.com/en/misc/mnglkbd.html maps & to
+### the Mongolian NIRUGU in the column for the English keyboard layout:
+&	᠊	0	### U+180A MONGOLIAN NIRUGU
+### The table on http://www.mongolfont.com/en/misc/mnglkbd.html maps ' to
+### the Mongolian NIRUGU in the column for the Japanese keyboard layout:
+'	᠊	0	### U+180A MONGOLIAN NIRUGU
+### The table on http://www.mongolfont.com/en/misc/mnglkbd.html maps / to
+### U+002E FULL STOP both in the column for the English and the Japanese keyboard layout:
+/	.	0	### U+002E FULL STOP
+### : is not used in http://www.mongolfont.com/en/misc/mnglkbd.html but
+### it seems to make sense to map it to the Mongolian colon:
+:	᠄	0	### U+1804 MONGOLIAN COLON
+### Cyrillic section, unfinished:
+ф	ᠹ	0	###
+ц	ᠴ	0	###
+у	ᠤ	0	###
+ж	ᠵ	0	###
+э	ᠡ	0	###
+н	ᠨ	0	###
+г	ᠭ	0	###
+ш	ᠱ	0	###
+ү	ᠦ	0	###
+з	ᠽ	0	###
+к	ᠻ	0	###
+ъ	ᠫ	0	###
+й	ᠩ	0	###
+ы	MVS	0	###
+б	ᠪ	0	###
+ө	ᠥ	0	###
+а	ᠠ	0	###
+х	ᠬ	0	###
+р	ᠷ	0	###
+о	ᠣ	0	###
+л	ᠯ	0	###
+д	ᠳ	0	###
+п	FVS1	0	###
+я	ᠶ	0	###
+ч	ᠼ	0	###
+ё	ᠧ	0	###
+с	ᠰ	0	###
+м	ᠮ	0	###
+и	ᠢ	0	###
+т	ᠲ	0	###
+ь	᠂	0	###
+в	᠃	0	###
+ю	NNBSP	0	###
+Б	ᠸ	0	###
+Р	ᠿ	0	###
+Т	᠅	0	###
+И	᠀	0	###
+Ь	《	0	###
+В	》	0	###
+Х	ᠾ	0	###
+Ч	ᡂ	0	###
+Я	ᡁ	0	###
+Л	ᡀ	0	###
+О	ᠺ	0	###
+Г	FVS2	0	###
+Ш	FVS3	0	###
+Ф	SWNJ	0	###
+У	【	0	###
+Ж	】	0	###
+Э	〈	0	###
+Н	〉	0	###
 END_TABLE
-- 
2.31.1


```

**Comment by mike-fabian** [comment]
I have sent you a merge request:

https://framagit.org/popolon/ibus-table-mongol_bichig/-/merge_requests/1

**Comment by mike-fabian** [comment]
🏓 Ping

**Comment by Popolon** [comment]
Thanks for your help merge done, sorry to answer lately.

**Comment by mike-fabian** [comment]
Great, no problem! I would like to fix the cyrillic section as well but so far I couldn’t find a reference either.

If we cannot find a reference for that, I’ll add it as it is now soon to ibus-table-others:

https://github.com/moebiuscurve/ibus-table-others

ibus-table-others is a collection of tables to be used with ibus-table except for the tables which are for  Chinese, the Chinese tables are in

https://github.com/definite/ibus-table-chinese

**Comment by Popolon** [comment]
I just think, that a Mongolian made an ibus-table entry for ibus called Maral, that I use sometimes to type mongolian cyrillic, it could be added if the author agree in ibus-table package, to have a more complete mongolian set, and he can perhaps help for this part, i made him suggestion few years ago too, to have his ibus-table working with my French keyboard.

https://github.com/ochko/ibus-table-mongolian

**Comment by WhiteCzar** [comment]
What kind of reference for the Cyrillic are you referring to? I speak Russian and learned a little Mongolian (traditional script), so I know both alphabets and can look up info in Russian if it helps. Though I cannot code.

**Comment by Popolon** [comment]
> What kind of reference for the Cyrillic are you referring to? I speak Russian and learned a little Mongolian (traditional script), so I know both alphabets and can look up info in Russian if it helps. Though I cannot code.

I read both too, but this is about the mapping of the mongolian characters on mongolian cyrillic keyboards (as used in Mongolia+Russia), as the reference was for a Latin keyboard (as used in China). Probably need people from Mongolia and or Kalmykia/Buriatia that know traditionnal Mongolian script to know how to place them the same way than for Latin one.

**Comment by mike-fabian** [comment]
> I just think, that a Mongolian made an ibus-table entry for ibus called Maral, that I use sometimes to type mongolian cyrillic, it could be added if the author agree in ibus-table package, to have a more complete mongolian set, and he can perhaps help for this part, i made him suggestion few years ago too, to have his ibus-table working with my French keyboard.
> 
> https://github.com/ochko/ibus-table-mongolian

This table from user “ochko” does something completely different, it is for typing Cyrillic on a Latin keyboard. It is not used for typing Mongolian characters. That could be added as well to ibus-table-others of course if this is considered useful, but it doesn’t help us completing the Cyrillic part of your mongol-bichig table.

**Comment by mike-fabian** [comment]
> > What kind of reference for the Cyrillic are you referring to? I speak Russian and learned a little Mongolian (traditional script), so I know both alphabets and can look up info in Russian if it helps. Though I cannot code.
> 
> I read both too, but this is about the mapping of the mongolian characters on mongolian cyrillic keyboards (as used in Mongolia+Russia), as the reference was for a Latin keyboard (as used in China). Probably need people from Mongolia and or Kalmykia/Buriatia that know traditionnal Mongolian script to know how to place them the same way than for Latin one.

Exactly!
We would need to know which Cyrillic key pressed on a Cyrillic keyboard layout should produce which Mongolian character.

**Comment by WhiteCzar** [comment]
I sent the link to this issue to my friend from Mongolia who also knows Trad Mongolian. He should be familiar with Cyrillic Mongolian keyboards in Mongolia. Perhaps he can reference something.

**Comment by mike-fabian** [comment]
> I sent the link to this issue to my friend from Mongolia who also knows Trad Mongolian. He should be familiar with Cyrillic Mongolian keyboards in Mongolia. Perhaps he can reference something.

Did you get a reply?

**Comment by WhiteCzar** [comment]
Unfortunately, I didn't. I guess he's busy. But I gave him the link, so if he has free time he will check this out.

**Comment by WhiteCzar** [comment]
Oh, I just asked him and he says he doesn't quite understand. @Popolon you can speak Mongolian right? Maybe you can type a message as to what exactly needs to be done? Because I'm not sure I will express it as well.

**Comment by WhiteCzar** [comment]
This is me trying to explain to him in my broken Chinese haha: 要知道，你用Cyrillic鍵盤打哪個字要變成哪個傳統蒙古文的字

**Comment by Popolon** [comment]
I can read/write Chinese a clumsy way, I only know some words and basic sentences of Mongolian, I learnt some base few wears ago but didn't practised since, forgot most of it. You wrote traditional Chinese, simplified chinese is used in Mainland china and so more probably in Mongolia neighbour. Even if lot of mainland Chinese (at least people in China for whom it is native language) can globally understand it and generally Taiwan/Hong Kong/Macau people (that use traditionnal) can read at least parts of simplified one. You can ask hom this, but it will be hard to go further :).

你能用传统蒙古文（Монгол бичиг/ᠮᠣᠩᠭᠣᠯ ᠪᠢᠴᠢᠭ）在西里尔文键盘（кирилээр бичсэн даруул）上打字吗？

I asked to a friend from Ulaanbaatar that lives here in Paris to help me to write it in Mongolian. She doesn't really know this script sadly. Next step ask also to Maral author that understand english.

**Comment by Popolon** [comment]
Forgot to copy past her translation:
> Oh, I just asked him and he says he doesn't quite understand. @Popolon you can speak Mongolian right? Maybe you can type a message as to what exactly needs to be done? Because I'm not sure I will express it as well.

туунеес би чамд бичий эгэлгуи яхав. За тур баяртаи

**Comment by WhiteCzar** [comment]
Yeah, he hasn't replied to me at all about this after I've sent him this message. I guess he's not interested in helping out with this. Really sorry.

**Comment by mike-fabian** [comment]
Included in ibus-table-others-1.3.12 now:

https://github.com/moebiuscurve/ibus-table-others/releases/tag/1.3.12
https://github.com/mike-fabian/ibus-table-others/releases/tag/1.3.12
